sdk/test-utils/recorder/test/node/recorder.spec.ts

```diff
@@ -0,0 +1,278 @@
+    expect(stripNewLines(recordingWithoutDate).replace(/[^a-zA-Z0-9]+/g, " ")).to.equal(
+      stripNewLines(expectedRecording).replace(/[^a-zA-Z0-9]+/g, " ")
+    );
+  });
+
+  it("should playback a simple test", async function() {
+    // The recorder will assume that it is in playback mode by default.
+    // process.env.TEST_MODE = "playback";
+
+    // Making sure the expected recording actually exists before running playback.
+    const fs = require("fs");
+    fs.writeFileSync(
+      "./recordings/node/the_recorders_public_api_on_nodejs/recording_should_playback_a_simple_test.js",
+      expectedRecording
+    );
+
+    // The recorder should start in the beforeEach call.
+    // To emulate that behavior while keeping the test code as contained as possible,
+    // we're compensating with this.
+    const fakeThis: TestContextInterface = new TestContext(this.test! as TestContextTest, {
+      ...this.currentTest,
+      file: __filename,
+      fn: emptyFunction
+    });
+
+    const recorder = record(fakeThis, recorderEnvSetup);
+    const response = await helloWorldRequest();
+
+    // The playback code served the appropriate response based on the recordings.
+    expect(response).to.equal(expectedHttpResponse);
+
+    recorder.stop();
+  });
+
+  it("soft-record should re-record a simple outdated test", async function() {
+    process.env.TEST_MODE = "soft-record";
+    process.env.SERVER_ADDRESS = "http://127.0.0.1:8080";
+
+    // Making sure the expected recording actually exists before running playback.
+    const fs = require("fs");
+    fs.writeFileSync(
+      "./recordings/node/the_recorders_public_api_on_nodejs/recording_softrecord_should_rerecord_a_simple_outdated_test.js",
+      expectedRecording
+    );
+
+    // We create a very simple HTTP server that serves some content at a specific port.
+    const http = require("http");
+    const server = http.createServer(function(_: any, res: any) {
+      res.write(expectedHttpResponse);
+      res.end();
+    });
+    server.listen(8080);
+
+    const changedTestFunction = () => {
+      let the_contents_have_changed = true;
+      return the_contents_have_changed;
+    };
+
+    // The recorder should start in the beforeEach call.
+    // To emulate that behavior while keeping the test code as contained as possible,
+    // we're compensating with this.
+    const fakeThis: TestContextInterface = new TestContext(this.test! as TestContextTest, {
+      ...this.currentTest,
+      file: __filename,
+      fn: changedTestFunction
+    });
+
+    const recorder = record(fakeThis, recorderEnvSetup);
+
+    const response = await helloWorldRequest();
+
+    // This test's request reached the server and received the expected response.
+    expect(response).to.equal(expectedHttpResponse);
+
+    // Cleaning everything before we continue verifying the results.
+    server.close();
+    recorder.stop();
+
+    // The recorder takes some time to finish writing the output file.
```

Comment:
is there a way we can get a promise back to await here? maybe stop can return a promise that completes when the writing is done?

Reply:
I can do that!

Reply:
I made an issue: https://github.com/Azure/azure-sdk-for-js/issues/7386